src/index/bwtree_index.cpp

```diff
@@ -106,19 +116,18 @@ BWTREE_INDEX_TYPE::DeleteEntry(const storage::Tuple *key,
 BWTREE_TEMPLATE_ARGUMENTS
 bool
 BWTREE_INDEX_TYPE::CondInsertEntry(const storage::Tuple *key,
-                                   const ItemPointer &location,
+                                   ItemPointer *location,
                                    std::function<bool(const ItemPointer &)> predicate) {
   KeyType index_key;
   index_key.SetFromKey(key);
   
-  ItemPointer *item_p = new ItemPointer{location};
   bool predicate_satisfied = false;
```

Comment:
itempointer is now allocated outside of this function and the pointer is passed to the function as a parameter. This simplifies memory reclamation.